projects/extension/src/background/ConnectionManager.ts

```diff
@@ -193,31 +193,45 @@ export class ConnectionManager extends (EventEmitter as {
 
   /** Handles the incoming message that contains Spec. */
   async #handleSpecMessage(
-    chainConnection: ChainConnection,
+    tab: Tab,
+    chainId: string,
     chainSpec: string,
     potentialRelayChainIds: string[],
   ) {
     if (!chainSpec) {
       throw new Error("Relay chain spec was not found")
     }
 
+    const chainConnection: ChainConnection = {
+      chainId,
+      tabId: tab.port.sender!.tab!.id!,
+      healthChecker: smHealthChecker(),
+      url: tab.port.sender!.url!,
+      port: tab.port,
+      chainName: JSON.parse(chainSpec).name.toLowerCase() || "unknown",
+    }
+
+    // TODO: should check for duplicates? or should the content script do that?
```

Comment:
The content-script doesn't have the means for checking for duplicates, b/c it doesn't have the full picture (an iframe would have a different content-script). So, it should be the background the one checking for duplicates.

Also, what IMO should happen if there was a duplicate is that the background should send an error message for that chainId, so that the existing chainConnection and the one that was possibly trying to impersonate the existing one get removed from the equation.